contracts/ConverterV2.sol

```diff
@@ -0,0 +1,198 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.7.0;
+
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "./polydex/interfaces/IPolydexERC20.sol";
+import "./polydex/interfaces/IPolydexPair.sol";
+import "./polydex/interfaces/IPolydexFactory.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+import "./polydex/interfaces/IPolydexRouter.sol";
+
+// ConverterV2 is Farm's left hand and kinda a wizard. He can create up CNT from pretty much anything!
+// This contract handles "serving up" rewards for xCNT holders & also burning some by trading tokens collected from fees for CNT.
+// This contract differs in the way from its predecessor by directly using the Polydex Router to swap LP tokens and ERC20 tokens to CNT.
+
+contract ConverterV2 is Ownable, ReentrancyGuard {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+    
+
+    IPolydexFactory public factory;
```

Comment:
Lets add functions to set polydex factory and WMatic as well. Only Owner functions